cmd/xl-v1-object.go

```diff
@@ -680,8 +680,7 @@ func (xl xlObjects) PutObject(bucket string, object string, size int64, data io.
 	}
 
 	// Rename the successfully written temporary object to final location.
-	onlineDisks, err = renameObject(onlineDisks, minioMetaTmpBucket, tempObj, bucket, object, xl.writeQuorum)
-	if err != nil {
+	if _, err = renameObject(onlineDisks, minioMetaTmpBucket, tempObj, bucket, object, xl.writeQuorum); err != nil {
```

Comment:
I don't encourage removing the update of onlineDisks here and in other places since we can forget this stuff when someone introduces new code after this line, it is better if we keep onlineDisks always updated.